### ISSUE TITLE
SDL3: camera-read-and-draw example broken output

### ISSUE THREAD

**Issue by ichordev** [libsdl-org/SDL#11372]
Having this problem on macOS 14.5 (M1), using both SDL 3.1.3 and [this commit](https://github.com/libsdl-org/SDL/commit/014b473bcc392cf8b1e83959602f973cb3afebf3) from just now.
Here's a screen-grab of a while envelope:
![image](https://github.com/user-attachments/assets/b931e462-2953-4af6-b525-6b06bd3cf27d)

It seems to be caused by `SDL_AcquireCameraFrame` returns because I really couldn't find any workaround. I tried manually passing various pixel formats to `SDL_CreateTexture`, but the only pixel format that didn't produce funny colour-interlacing (like in the screenshot above), or multiple side-by-side technicolour/greyscale images, was `SDL_PIXELFORMAT_RGB332`. Instead, it produces *one* technicolour image:
![Untitled](https://github.com/user-attachments/assets/a02f7900-9bce-4783-aa75-3f07dd039b86)
Reminder: this is a white object.

Could possibly be caused by #9504?

**Comment by slouken** [comment]
Does testcamera work? If not, what is the output of testcamera?

**Comment by ichordev** [comment]
> Does testcamera work? If not, what is the output of testcamera?

Tested just now with SDL 3.1.6 and [this commit](https://github.com/libsdl-org/SDL/commit/e2323c136744ce5fe60800fb359257b268bde4a5) (main). Same system and OS version. Has the exact same problem. Here's the output (sans any times):
```
test[63308:9663788] Saw 2 camera devices.
test[63308:9663788]   - Camera #0:  KC#43157174
test[63308:9663788]   - Camera #1:  OBS Virtual Camera
test[63308:9663788] Available formats:
test[63308:9663788]     2592x1944 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     2592x1944 2.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     2560x1440 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     2560x1440 2.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     1920x1080 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     1920x1080 2.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     1600x1200 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     1600x1200 5.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     1280x960 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     1280x960 5.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     1280x720 60.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     1280x720 7.50 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     960x720 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     960x720 15.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     800x600 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     800x600 15.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788]     640x480 60.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788]     640x480 30.00 FPS SDL_PIXELFORMAT_YUY2
test[63308:9663788] Camera approved!
test[63308:9663788] Camera Spec: 2592x1944 30.00 FPS SDL_PIXELFORMAT_NV12
test[63308:9663788] Quit!
```
Output was identical for both versions tested.

**Comment by slouken** [comment]
Great, can you add this diff and send the output?
```diff
diff --git a/src/camera/coremedia/SDL_camera_coremedia.m b/src/camera/coremedia/SDL_camera_coremedia.m
index 800340b4d..b01149178 100644
--- a/src/camera/coremedia/SDL_camera_coremedia.m
+++ b/src/camera/coremedia/SDL_camera_coremedia.m
@@ -211,6 +211,7 @@ static SDL_CameraFrameResult COREMEDIA_AcquireFrame(SDL_Camera *device, SDL_Surf
                 size_t plane_height = CVPixelBufferGetHeightOfPlane(image, i);
                 size_t plane_pitch = CVPixelBufferGetBytesPerRowOfPlane(image, i);
                 size_t plane_size = (plane_pitch * plane_height);
+SDL_Log("Plane %d: src = %p, height = %d, pitch = %d\n", i, src, (int)plane_height, (int)plane_pitch);
                 SDL_memcpy(dst, src, plane_size);
                 dst += plane_size;
             }
```

**Comment by ichordev** [comment]
> Great, can you add this diff and send the output?

While trying to make this patch do anything I realised that I've only been runtime-linking against the version of SDL3 installed in my system (previously 3.1.3, and now 3.1.6), and never against the most recent commit. Upon *actually* testing against [main](https://github.com/libsdl-org/SDL/commit/49153ebfc4cd857769777aa5a2edd3aa5bbd8304), I can confirm that this issue has been resolved for me with both `camera-read-and-draw` and `testcamera`. The output of `testcamera` is still exactly the same. Sorry for the error on my part, and thank you for getting this fixed.
